content/en/blog/_posts/2023-xx-xx-openapi-v3-field-validation-ga.md

```diff
@@ -0,0 +1,107 @@
+of the OpenAPI and is an improvement upon [OpenAPI
+V2](https://kubernetes.io/blog/2016/12/kubernetes-supports-openapi/)
+which has been supported since Kubernetes 1.5. OpenAPI V3 support was
+added in Kubernetes in v1.23, moved to beta in v1.24 and is now GA in
+v1.27.
+
+## OpenAPI V3
+
+### What does OpenAPI V3 offer over V2
+
+#### Built-in types
+
+Kubernetes offers certain annotations on fields that are not
+representable in OpenAPI V2, or sometimes not represented in the
+OpenAPI v2 that Kubernetes generate. Most notably, the "default" field
+is published in OpenAPI V3 while omitted in OpenAPI V2. A single type
+that can represent multiple types is also expressed correctly in
+OpenAPI V3 with the oneOf field. This includes proper representations
+for IntOrString and Quantity.
+
+#### Custom Resource Definitions
+
+In Kubernetes, Custom Resource Definitions use a structural OpenAPI V3
+schema that cannot be represented as OpenAPI V2 without a loss of
+certain fields. Some of these include nullable, default, anyOf, oneOf,
+not, etc. OpenAPI V3 is a completely lossless representation of the
+CustomResourceDefinition structural schema.
+
+### How do I use it?
+
+The OpenAPI V3 root discovery can be found at the `/openapi/v3`
+endpoint of a Kubernetes cluster. OpenAPI V3 documents are grouped by
+group-version to reduce the size of the data transported, the separate
+documents can be accessed at `/openapi/v3/apis/<group>/<version>` and
+`/openapi/v3/api/v1` representing the legacy group version. Please
+refer to the [Kubernetes API
+Documentation](https://kubernetes.io/docs/concepts/overview/kubernetes-api/)
+for more information around this endpoint.
+
+Various consumers of the OpenAPI have already been updated to consume v3, including the entirety of kubectl, and server side apply. An OpenAPI V3 Golang client is available in client-go. Please refer to the [interface](https://github.com/kubernetes/client-go/blob/release-1.27/openapi3/root.go) for details on usage.
+
+## Server Side Field Validation
+
+The query parameter `fieldValidation` may be used to indicate the level of field validation the server should perform. If the parameter is not passed, server side field validation is in Warn mode by default.
+
+- Strict: Strict field validation, errors on validation failure
+- Warn: Field validation is performed, but errors are exposed as warnings rather than failing the request
+- Ignore: No server side field validation is performed
+
+kubectl will skip client side validation and will automatically use server side field validation in strict mode. Controllers by default use server side field validation in warn mode.
+
+With client side validation, we had to be extra lenient because some fields were missing from OpenAPI V2 and we didn’t want to reject possibly valid objects. This is all fixed in server side validation. Additional documentation may be found [here](https://kubernetes.io/docs/reference/using-api/api-concepts/#field-validation)
```

Comment:
```suggestion
With client side validation, we had to be extra lenient because some fields were missing
from OpenAPI V2 and we didn’t want to reject possibly valid objects.
This is all fixed in server side validation. Additional documentation may be found
[here](/reference/using-api/api-concepts/#field-validation)
```

Reply:
Just want to confirm, this URL should be `/docs/reference/using-api/api-concepts/#field-validation` instead of `/reference/using-api/api-concepts/#field-validation`?